### ISSUE TITLE
Bug: `key-spacing` with `align: "value"` produces incorrect alignment

### ISSUE THREAD

**Issue by fasttime** [eslint/eslint#16490]
### Environment

Node version: v19.0.0
npm version: v8.19.2
Local ESLint version: v8.26.0
Global ESLint version: Not found
Operating System: darwin 22.1.0

### What parser are you using?

Default (Espree)

### What did you do?

```js
/* eslint "key-spacing": ["error", { align: "value" }] */
var foo =
{
  code: 1,
  message:
  "some value on the next line",
};
```

[**DEMO**](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IFwia2V5LXNwYWNpbmdcIjogW1wiZXJyb3JcIiwgeyBhbGlnbjogXCJ2YWx1ZVwiIH1dICovXG52YXIgZm9vID1cbntcbiAgY29kZTogMSxcbiAgbWVzc2FnZTpcbiAgXCJzb21lIHZhbHVlIG9uIHRoZSBuZXh0IGxpbmVcIlxufTsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

### What did you expect to happen?

No error.

### What actually happened?

Got error
> 4:9  Missing space before value for key 'code'.  ([key-spacing](https://eslint.org/docs/rules/key-spacing))

Autofix adds three spaces after `code:` to align the value `1` one space after `message:`, which doesn't seem correct since there is no other value to align with.

If the order of the properties is swapped, as expected, no error is reported.

```js
/* eslint "key-spacing": ["error", { align: "value" }] */
var foo =
{
  message:
  "some value on the next line",
  code: 1
};
```

[**DEMO**](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IFwia2V5LXNwYWNpbmdcIjogW1wiZXJyb3JcIiwgeyBhbGlnbjogXCJ2YWx1ZVwiIH1dICovXG52YXIgZm9vID1cbntcbiAgbWVzc2FnZTpcbiAgXCJzb21lIHZhbHVlIG9uIHRoZSBuZXh0IGxpbmVcIixcbiAgY29kZTogMVxufTsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
This looks like a bug to me.

> ```js
> /* eslint "key-spacing": ["error", { align: "value" }] */
> var foo =
> {
>   code: 1,
>   message:
>   "some value on the next line",
> };
> ```
> If the order of the properties is swapped, as expected, no error is reported.
> 
> ```js
> /* eslint "key-spacing": ["error", { align: "value" }] */
> var foo =
> {
>   message:
>   "some value on the next line",
>   code: 1
> };
> ```

The different behavior seems to be because in the first case [`continuesPropertyGroup`](https://github.com/eslint/eslint/blob/74a5af487ac7296a46a8078e585f00df72b63d83/lib/rules/key-spacing.js#L343) considers the two properties to be in the same group, while in the second case they are in separate groups.

https://github.com/eslint/eslint/blob/74a5af487ac7296a46a8078e585f00df72b63d83/lib/rules/key-spacing.js#L343-L349

Perhaps we could change this logic a bit to fix the bug - if `value` is not on the same line as `key`, then property doesn't continue the group?

**Comment by fasttime** [comment]
Thanks @mdjermanovic, I did some debugging and I think your analysis is correct. I can work on a PR to fix this.
It seems that there are currently to [unit tests](https://github.com/eslint/eslint/blob/c3ce5212f672d95dde3465d7d3c4bf99ff665f8b/tests/lib/rules/key-spacing.js) covering the case of a property where the value starts on a different line than the key, so I guess I will add some.

**Comment by snitin315** [comment]
@fasttime Thanks for taking this up. Yes, you will have to add corresponding test cases in your PR.

**Comment by fasttime** [comment]
> Perhaps we could change this logic a bit to fix the bug - if `value` is not on the same line as `key`, then property doesn't continue the group?

This should be fine with `align: "value"`. I'm not sure about `align: "colon"`. So, for instance, this code is currently considered ok by the rule:

```js
/* eslint "key-spacing": ["error", { align: "colon" }] */
({
  command  : "npm",
  arguments:
  [
    "test"
  ]
});
```

But if we change the logic such that a value on a different line doesn't continue the group, then the rule would report a problem, and the corrected output would be:

```js
/* eslint "key-spacing": ["error", { align: "colon" }] */
({
  command: "npm",
  arguments:
  [
    "test"
  ]
});
```

Any thoughts about this @mdjermanovic?
